internal/cloud/auth/auth.go

```diff
@@ -47,7 +49,16 @@ func Login(ctx context.Context, out io.Writer) error {
 		return err
 	}
 
-	fmt.Fprintf(out, "\nTo authenticate, visit:\n\t%s\n\n", deviceAuth.VerificationURIComplete)
+	authURL := deviceAuth.VerificationURIComplete
+
+	browserBuf := new(strings.Builder)
+	browser.Stdout = browserBuf
+	browser.Stderr = browserBuf
+	if err := browser.OpenURL(authURL); err != nil {
+		fmt.Fprintf(out, "\nFailed to open browser: %s\n\n%s", err, browserBuf.String())
+	}
+
+	fmt.Fprintf(out, "\nTo authenticate, visit:\n\t%s\n\n", authURL)
```

Comment:
nit: should we print a message that indicates that the browser window has been opened?

Reply:
we're not printing the message in other places  so maybe we can just avoid it for consistency?

Reply:
Tweaked the wording:

```
⬢ [fedora-toolbox:38] ❯ dagger-dev login dagger
Browser opened to: https://auth.dagger.cloud/activate?user_code=XLVW-QSJR
Confirmation code: XLVW-QSJR

Success.
```

Here's how it looks when we failed to open the browser:

```
⬢ [fedora-toolbox:38] ❯ dagger-dev login dagger
Failed to open browser: oh no

Authenticate here: https://auth.dagger.cloud/activate?user_code=GFZT-DZXD
Confirmation code: GFZT-DZXD
```

Also the confirmation code is in bold:

![image](https://github.com/user-attachments/assets/335d67a1-fe0f-4abb-905a-5e59baeb057d)

We only showed it in the URL before which seemed a bit subtle considering the landing page tells you to confirm it.

Reply:
SGTM :heavy_check_mark: